matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/pushers/DefaultPushersService.kt

```diff
@@ -95,7 +107,14 @@ internal class DefaultPushersService @Inject constructor(
         return request.id
     }
 
-    override suspend fun removeHttpPusher(pushkey: String, appId: String) {
+    private fun JsonPusher.validateParameters() {
+        // Do some parameter checks. It's ok to throw Exception, to inform developer of the problem
+        if (this.pushKey.length > 512) throw InvalidParameterException("pushkey should not exceed 512 chars")
+        if (this.appId.length > 64) throw InvalidParameterException("appId should not exceed 64 chars")
+        this.data?.url?.let { url -> if ("/_matrix/push/v1/notify" !in url) throw InvalidParameterException("url should contain '/_matrix/push/v1/notify'") }
```

Comment:
Generally, if it is not necessary, I do not like to explicitly use `this.`. We can discuss it if you want to :)

Reply:
happy to remove :+1: